### PR TITLE
Update sidebar tab design

### LINKED ISSUE #6056: Improve the look of Sidebar tabs
## Context
- https://www.notion.so/pixiebrix/Improve-the-look-of-our-Sidebar-tabs-f46e3cc098cf4e749d8819e7da90ef4e

## Scope
- [ ] Ensure tabs remain in a single row
- [ ] Tabs evenly fill all available width (replace `fill` with `justify`)
- [ ] Handle Tab title overflow
- [ ] Replace the icon used for close buttons
- [ ] Hover state for close button (background color and border radius)
- [ ] Hover state for close button of inactive tab sets hover state for tab
- [ ] Border rounding on active tabs
- [ ] Border rounding and backgroundColor on tabs when in `hover` state
  - [ ] Sibling on left should have border-bottom-right rounding if not in active state
  - [ ] Sibling on right should always have border-bottom-left rounding if not in active state
  - [ ] Background color outside the border-radius should be hover color
- [ ] Border rounding on siblings of active tabs
  - [ ] Sibling on left should always have border-bottom-right rounding
  - [ ] Sibling on right should always have border-bottom-left rounding
  - [ ] Background color outside the border-radius should be active color
- [ ] Divider between inactive & unhovered tabs (potentially use an :after)
- [ ] Hide tabs after 8 have been added, [(see notion comment)](https://www.notion.so/Improve-the-look-of-our-Sidebar-tabs-f46e3cc098cf4e749d8819e7da90ef4e?d=1c0795a3d4b140f6a60c4d481b0cc334&pvs=4#ab2a784a23244363a938a0326191ccdb)

## Notes
- All tabs have the `nav-link` css class and the active tab has `active`. This should make it easy to style the tabs using the adjacent sibling combinator, nth-child pseudo class, etc.

### PR DESCRIPTION
**Pull request by turbochef**
## What does this PR do?

- Closes #6056 
- Add missing design system colors to `colors.scss`
- Fix active state class
- Update tab design
  - [x] Ensure tabs remain in a single row
  - [x] Tabs evenly fill all available width (replace `fill` with `justify`)
  - [x] Handle Tab title overflow
  - [x] Replace the icon used for close buttons
  - [x] Hover state for close button (background color and border radius)
  - [x] Hover state for close button of inactive tab sets hover state for tab
  - [x] Border rounding on active tabs
  - [x] Border rounding and backgroundColor on tabs when in `hover` state
    - [ ] Sibling on left should have border-bottom-right rounding if not in active state
    - [ ] Sibling on right should always have border-bottom-left rounding if not in active state
    - [ ] Background color outside the border-radius should be hover color
  - [x] Border rounding on siblings of active tabs
    - [ ] Sibling on left should always have border-bottom-right rounding
    - [ ] Sibling on right should always have border-bottom-left rounding
    - [ ] Background color outside the border-radius should be active color
  - [x] Hide tabs after 8 have been added
  - [x] Divider between inactive & unhovered tabs (potentially use an :after)
 
~~Because the `Nav.Link` component now has a border radius, `border-right` looks like a shadow.`::after` shows the divider inside the tab rather than next to it. But because bootstrap does a lot of positioning on `Nav.Link` I can't simply give it a container div.~~

By I got around the bootstrap issues by wrapping all the Nav.Links in a `Nav.Item`.

## Demo

- [loom](https://www.loom.com/share/f43b7fab21324e668b9c70e951344510?sid=e81f8414-1628-4875-b1ac-b065dee1ba53)

Max tabs while hovering on the 3rd:
image

3 tabs:
image

Single tab:
image

## Checklist

- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/sidebar/Tabs.tsx [hunk 1]
```diff
@@ -99,6 +100,18 @@ const TemporaryPanelTabPane: React.FC<{
 });
 TemporaryPanelTabPane.displayName = "TemporaryPanelTabPane";
 
+const TabWithDivider = ({ children, active, ...props }: NavLinkProps) => (
+  <Nav.Item className={cx(styles.tabWrapper, active && styles.active)}>
```
Comment (twschiller) [inline src/sidebar/Tabs.tsx:104]: I think this should use the object arg form:

```
cx(styles.tabWrapper, {[styles.active]: active})}
```

File: src/sidebar/Tabs.tsx [hunk 2]
```diff
@@ -132,72 +145,81 @@ const Tabs: React.FC = () => {
...
     <Tab.Container
       id="panel-container"
       defaultActiveKey={activeKey}
       activeKey={activeKey}
     >
-      <div className="full-height bg-white">
-        <Nav fill variant="tabs" onSelect={onSelect}>
+      <div className="full-height">
+        <Nav
+          fill
```
Comment (grahamlangford) [inline src/sidebar/Tabs.tsx:159]: This prop should be changed to `justify`. You should be able to simplify `.tabContainer` after that.

File: src/sidebar/Tabs.tsx (hunk 1 above)
Comment (grahamlangford) [reply src/sidebar/Tabs.tsx:104]: Doesn't the `Nav.Item` have the `.active` class? You should be able to write selectors that don't require you to worry about the active prop.

Comment (turbochef) [reply src/sidebar/Tabs.tsx:104]: @twschiller That's fine, I'll update it.
@grahamlangford The selector gets transformed so we can't use `.tabWrapper{ &.active{...}}`.

**Comment by grahamlangford** [comment]
Can you add a loom demoing the tabs in the various states?

**Comment by turbochef** [comment]
@grahamlangford I updated the loom in the description

**Comment by grahamlangford** [comment]
Missing the border radius at the bottom of the tabs:
image
image
image

**Comment by turbochef** [comment]
@grahamlangford Sorry for missing that. I've update the code, screenshots, and video to include the outer corner decorations.

**Comment by grahamlangford** [comment]
> @grahamlangford Sorry for missing that. I've update the code, screenshots, and video to include the outer corner decorations.

I think you still missed the outer corner decorations for the `hover` state. Should be both sides unless it's directly adjacent to the `active` tab.

**Comment by grahamlangford** [comment]
Also, in the loom, the hover effect on the close button looks wrong. Can you post a screenshot?

**Comment by turbochef** [comment]
@grahamlangford updated the close button to not overlap with the corner decoration. Updated screenshots.

**Comment by twschiller** [comment]
FWIW, this would be a good PR to add to Storybook. (And developing against Storybook could help speed up dev vs. having to reload sidebar during testing)

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**